github3/gists/file.py

```diff
@@ -31,7 +33,8 @@ def _update_attributes(self, attributes):
         #: The size of the file.
         self.size = attributes.get('size')
         #: The content of the file.
-        self.content = attributes.get('content')
+        r = requests.get(self.raw_url)
```

Comment:
@ueg1990 to answer the question you asked, do not conditionally make an extra request.

Do the following

- In `_update_attributes` do:

  ```py
  self.original_content = attributes.get('content')
  ```

- Add a method named `content(self)` to the `GistFile` class that looks like:

  ```py
  def content(self):
      resp = self._get(self.raw_url)
      if self._boolean(resp, 200, 404):
          return resp.content
      return None
```